datahub-web-react/src/app/tags/TagsTable.tsx

```diff
@@ -0,0 +1,436 @@
+                },
+            },
+            {
+                title: 'Owners',
+                key: 'owners',
+                render: (record) => {
+                    return <TagOwnersCell key={`owners-${record.entity.urn}`} tagUrn={record.entity.urn} />;
+                },
+            },
+            {
+                title: 'Applied to',
+                key: 'appliedTo',
+                render: (record) => {
+                    return <TagAppliedToCell key={`applied-${record.entity.urn}`} tagUrn={record.entity.urn} />;
+                },
+            },
+            {
+                title: '',
+                key: 'actions',
+                alignment: 'right' as AlignmentOptions,
+                render: (record) => {
+                    const items = [
+                        {
+                            key: '0',
+                            label: (
+                                <MenuItem
+                                    onClick={() => {
+                                        // Open color picker modal/drawer
+                                        setTagToEditColor(record.entity.urn);
+                                        setShowColorPicker(true);
+                                    }}
+                                    data-testid="action-edit"
+                                >
+                                    Edit
+                                </MenuItem>
+                            ),
+                        },
+                    ];
+
+                    return (
+                        <CardIcons>
+                            <Dropdown
+                                menu={{ items }}
+                                trigger={['click']}
+                                data-testid={`${record.entity.urn}-actions-dropdown`}
+                            >
+                                <Icon icon="MoreVert" size="md" />
+                            </Dropdown>
+                        </CardIcons>
+                    );
+                },
+            },
+        ],
+        [entityRegistry, searchQuery, sortedInfo],
+    );
+
+    // Generate table data once with memoization
+    const tableData = useMemo(() => {
+        return filteredTags.map((tag) => ({
+            ...tag,
+            key: tag.entity.urn,
+        }));
+    }, [filteredTags]);
+
+    return (
+        <>
+            <Table
+                columns={columns}
+                data={tableData}
+                isLoading={isLoading}
+                isScrollable
+                rowKey="key"
+                onChange={handleTableChange as any}
+            />
+
+            {showColorPicker && (
+                <ManageTag
+                    tagUrn={tagToEditColor}
+                    onClose={() => setShowColorPicker(false)}
+                    onSave={refetch}
+                    isModalOpen={showColorPicker}
+                />
+            )}
```

Comment:
so is this editing a tag? can you do anything other than edit the color? if so maybe instead of `showColorPicker` we say `tagToEdit` or something like that

Reply:
yeah it's really about editing good idea

Reply:
refactored!